### PR TITLE
[MM-20414] Migrated NavbarInfoButton to Typescript

### PR DESCRIPTION
**Pull request by Saucistophe**
#### Summary
Migrate Navigation bar Info Button to Typescript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16159
https://mattermost.atlassian.net/browse/MM-20414

### CODE REVIEW

**Review by hahmadia — COMMENTED** [review]
@Saucistophe 

Code looks good! 

I guess my only question is why in some places was `ChannelInfoButton` renamed to `NavbarInfoButton` ? 
Why couldn't we just keep the name `ChannelInfoButton` as it was? 

I'm not 100% understanding those changes. Thanks!

**Comment by Saucistophe** [comment]
> I guess my only question is why in some places was `ChannelInfoButton` renamed to `NavbarInfoButton` ?
> Why couldn't we just keep the name `ChannelInfoButton` as it was?

The class is named NavbarInfoButton, so it felt wrong importing `NavbarInfoButton` as `ChannelInfoButton` in the tests. I'm not sure which one is the best :shrug:

File: components/channel_header_mobile/channel_info_button/channel_info_button.tsx [hunk 1]
```diff
@@ -15,28 +17,32 @@ import InfoIcon from 'components/widgets/icons/info_icon';
...
+    hide: () => void;
+}
+
+export interface NavbarInfoButtonProps {
+    channel: Channel,
+    isReadOnly: boolean,
+    isRHSOpen?: boolean,
+    currentRelativeTeamUrl?: string,
+    actions: {
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => void;
```
Comment (calebroseland) [inline components/channel_header_mobile/channel_info_button/channel_info_button.tsx:30]: This one seems different than the one in the index? 

```diff
- => void
+ => Promise<{data: boolean}>
```

File: components/channel_header_mobile/channel_info_button/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import {openModal} from 'actions/views/modals';
+
+import ChannelInfoButton from './channel_info_button';
+
+type Actions = {
+    openModal: (modalData: {modalId: string; dialogType: React.Component; dialogProps?: any}) => Promise<{
+        data: boolean;
+    }>;
+}
```
Comment (calebroseland) [inline components/channel_header_mobile/channel_info_button/index.ts:17]: One option could be to use `ComponentProps<T>` to extract the types from the component itself. 0/5
```tsx
type Actions = ComponentProps<typeof ChannelInfoButton>['actions'];
```

**Review by calebroseland — CHANGES_REQUESTED** [review]
> > I guess my only question is why in some places was `ChannelInfoButton` renamed to `NavbarInfoButton` ?
> > Why couldn't we just keep the name `ChannelInfoButton` as it was?
> 
> The class is named NavbarInfoButton, so it felt wrong importing `NavbarInfoButton` as `ChannelInfoButton` in the tests. I'm not sure which one is the best 🤷

@Saucistophe I guess the file was originally named `navbar_info_button`, but got renamed to `channel_info_button` somewhere along the way. I agree having a consistent file/classname is best. 

My vote would be to use `ChannelInfoButton` where possible, possibly even changing `.navbar-info-button` to `.channel-info-button`  in the HTML. (I'm not seeing `.navbar-info-button` used in CSS or selectors elsewhere.)

**Comment by devinbinnie** [comment]
Closing this PR as inactive, @Saucistophe let us know if you plan on finishing this up.

**Pull request closed without merging**